### DISCUSSION TITLE
"No workspaces found" when generating signing key.

### DISCUSSION THREAD

**Discussion by privatedev11** [tauri-apps/tauri#6007] [Q&A] [answered]
When I run ``tauri signer generate -w ~/.tauri/myapp.key`` to generate a key, I get the following:

```
npm ERR! No workspaces found:
npm ERR!   --workspace=~/.tauri/myapp.key

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\USER\AppData\Local\npm-cache\_logs\2023-01-07T13_16_19_707Z-debug-0.log
```
username has been removed for privacy reasons

How can I fix this? Am I doing something wrong?
I am quite new to Tauri and Node.js so apologies if it is a noob error.

**Comment by FabianLars** [comment, marked as answer]
npm needs `--` to forward the args to tauri, like `npm run tauri signer generate -- -w ~/.tauri/myapp.key`
Also note that `~/` doesn't work on Windows, it will instead create a literal `~` folder in your current directory

p.s. i'm in the process of rewriting the guide, including these parts to hopefully make it easier & more failproof.

**Reply by privatedev11** [reply]
Thank you, that has worked.

**Comment by Emmanuel97423** [comment]
Hi! i've same issue on windows:
```
npm run tauri signer generate -- -w $HOME/.tauri/myapp.key

npm run tauri signer generate -- -w $HOME/.tauri/myapp.key
npm ERR! No workspaces found:
npm ERR!   --workspace=C:\Users\Epokw/.tauri/myapp.key

```
Someone can help me ? :)

**Reply by FabianLars** [reply]
works fine for me 🤔 
![grafik](https://github.com/tauri-apps/tauri/assets/30730186/1c3b8574-a2f0-4f49-8508-ddb9fa81206d)

Tested in powershell and git bash (that mingw64 thingy). What are you using?

Also, can you try this one?
```
npx @tauri-apps/cli signer generate -w $HOME/.tauri/myapp.key
```

**Reply by Emmanuel97423** [reply]
Yes ! Thank you its ok !
Can you explain me this issue ?

**Reply by FabianLars** [reply]
I have no idea tbh. Normally the extra -- is all you need. Maybe your specific npm version has an issue?

**Reply by Roalkege** [reply]
Hey I have the exact issue.
The npm run command does not work but the npx command works for me.

**Reply by domsleee** [reply]
It seems that two `--` are required in some installations of powershell, this command works for me:
> npm run tauri signer generate -- -- -w $HOME/.tauri/myapp.key

Explained in this comment: https://github.com/npm/cli/issues/6041#issuecomment-1610927104

If `npm` points to a `ps1` script, calling the `ps1` script strips away one of the `--`, before it calls the `npm` binary.

It can be checked with this command, if the output ends with `.ps1`, two `--` are required:
```powershell
❯ which npm
C:\Users\user\scoop\apps\nvs\current\nodejs\default\npm.ps1
```

**Reply by domsleee** [reply]
More info about powershell `npm.ps1` issues in https://github.com/npm/cli/issues/3136

**Comment by delebash** [comment]
npx worked for me as well on Windows 11.  Although, it did not create a file if I used ~.  I used `npx tauri signer generate -- -w key/.tauri/uvo.key`
 and it created a key folder in my root project with the key files.  I just moved the directory out of my project folder to another location

**Comment by chnkarl** [comment]
> npm run tauri signer generate -- -- -w $HOME/.tauri/myapp.key

double --, like -- --, worked tome.
